app/jinja_filters.py

```diff
@@ -311,6 +301,45 @@ def format_date_range_no_repeated_month_year(context, start_date, end_date, date
     return mark_safe(context, output)
 
 
+def _get_first_non_empty_item(items):
+    """
+    :param items: anything that is iterable
+    :return: first non empty value
+
+    In this filter the following values are considered false:
+    - None
+    - False
+    - any empty sequence, for example, '', (), [].
+    - any empty mapping, for example, {}.
+    - instances of user-defined classes, if the class defines a __bool__() method, when that method returns bool value False.
+
+     Note: to guarantee the returned element is actually the first non empty element in the iterable,
+    'items' must be a data structure that preserves order, ie tuple, list etc.
+    If order is not important, this can be reused to return `one of` the elements which is non empty.
+    """
+    for item in items:
+        if item:
+            return item
+
+        # The below statement will treat zero of any numeric type for example, 0, 0.0, 0j as a valid item since they are naturally 'falsy' in Python.
+        # Note: Booleans are a subtype of integers. Zero of any numeric type 'is not False' but 'equals False'.
+        # Reference: https://docs.python.org/release/3.4.2/library/stdtypes.html?highlight=boolean#boolean-values
+        if item is not False and item == 0:
```

Comment:
Splitting this into a separate if seems excessive. Also this wouldn't deal with a list of booleans (not that I can imagine it would ever need to). Wouldn't it be simpler to change this to:
```
    for item in items:
        if item or item is False or item == 0:
            return item

    raise Exception('No valid items provided.')
```